imkar/scattering/diffuse.py

```diff
@@ -0,0 +1,181 @@
+
+    speed_of_sound : array, float
+        Array containing the speed of sound in m/s for every measurement
+        condition calculated by the temperature and air humidity during
+        measurement time.
+
+    air_attenuation : array, pf.FrequencyData
+        Matrix or FrequencyData containing the air attenuation in 1/m for every
+        measurement condition calculated by temperature and air humidity during
+        measurement time. Its cshape has to be (..., 4).
+
+    calculation_method : 'ISO','Eyring' or 'Sabine'
+        Defines the method by which alpha is calculated.
+        For method 'ISO' the baseplate_area and sample_area need to be equal.
+
+    scale_factor : float
+        Factor that corrects the frequencies when using a smaller scale
+        reverbaration room than defined in the ISO standard. Standard value is
+        5, according to the measurement chamber at IHTA.
+
+    sample_area : float
+        Surface area of the measurement sample in square meter. Standard value
+        according to the test sample.
+
+    baseplate_area : float
+        Surface area of the baseplate in square meter. Standard value according
+        to the measurement chamber at IHTA.
+
+    surface_room : float
+        Surface area of the measurement room in square meter. Standard value
+        according to measurement chamber at IHTA.
+
+    volume_room : float
+        Volume of the empty measurement room in cubic meter. Standard value
+        according to measurement chamber at IHTA.
+
+    Returns
+    -------
+    scattering_coefficient : pf.FrequencyData
+        The calculated solution for the diffuse scattering coefficient after
+        International Standard 17497-1:2004(E) and the scattering coefficient
+        of the base plate.
+
+    References
+    ----------
+    .. [#] ISO 17497-1:2004, Sound-scattering properties of surfaces. Part 1:
+           Measurement of the random-incidence scattering coefficient in a
+           reverberation room. Geneva, Switzerland: International
+           Organization for Standards, 2004.
+    """
+    # check inputs
+    if reverberation_time.cshape[-1] != 4:
+        raise ValueError("reverberation_time.cshape has to be (..., 4)")
+
+    if (np.asarray(speed_of_sound, dtype=float).ndim == 1):
+        speed_of_sound = (np.asarray(speed_of_sound, dtype=float)[:, None]).T
+    else:
+        speed_of_sound = (np.asarray(speed_of_sound, dtype=float))
+
+    if (speed_of_sound.shape != reverberation_time.cshape):
+        raise ValueError("speed_of_sound has to be an array containing the \
+                         speed of sound during all 4 measurement conditions. \
+                         Its shape has to be (..., 4) and equal to \
+                         reverberation_time.cshape.")
+
+    if (np.any(np.asarray(speed_of_sound, dtype=float) <= 0)):
+        raise ValueError("speed_of_sound has to be a value bigger than zero")
+
```

Comment:
add a check if frequencies of ``air_attenuation``and ``reverberation_times``are same.